src/rgw/rgw_common.cc

```diff
@@ -767,102 +762,25 @@ std::string calc_hash_sha256_restart_stream(SHA256 **phash)
 
 int gen_rand_base64(CephContext *cct, char *dest, int size) /* size should be the required string size + 1 */
 {
-  char buf[size];
-  char tmp_dest[size + 4]; /* so that there's space for the extra '=' characters, and some */
+  vector<char> buf(size); 
+  vector<char> tmp_dest(size + 4); /* so that there's space for the extra '=' characters, and some */
```

Comment:
you've introduced vectors here and elsewhere - could we switch these to small_vector as well? a static size of 64 should cover the main uses